src/routing_table/prefix.rs

```diff
@@ -163,13 +163,21 @@ impl<T: Clone + Copy + Default + Binary + Xorable> PartialOrd<Prefix<T>> for Pre
         if self == other {
             Some(Ordering::Equal)
         } else if self.is_compatible(other) {
-            None
+            Some(self.bit_count().cmp(&other.bit_count()))
         } else {
             Some(self.name.cmp(&other.name))
         }
     }
 }
 
+impl<T: Clone + Copy + Default + Binary + Xorable> Ord for Prefix<T> {
+    fn cmp(&self, other: &Self) -> Ordering {
+        // safe - all paths in partial_cmp return Some()
+        // Option::expect used to satisfy Clippy
+        self.partial_cmp(other).expect("Prefix::partial_cmp returned None")
```

Comment:
I'd rather put the actual implementation here, and use `cmp` in `partial_cmp`. That way, no `expect` call is needed.